### PR TITLE
Fix List Examples #331

### PR DESCRIPTION
**Pull request by emilyrohrbough**
### Summary
This PR does two things:

1. Fixes the functionality of the nightwatch examples. For example, the default list displayed a blank page so context was added.
2. Moves edge-case examples from terra-site to nightwatch and adds the appropriate tests for the moved examples. 

This fix addresses issue #331

@cerner/terra

### CODE REVIEW

File: packages/terra-list/tests/nightwatch/multi-select-list/OnChangeMultiSelectList.jsx [hunk 1]
```diff
@@ -2,12 +2,14 @@ import React from 'react';
 
 import MultiSelectList from '../../../lib/MultiSelectList';
 
+/* eslint-disable no-alert */
 const list = () => (
-  <MultiSelectList onChange={event => event.delegateTarget} >
+  <MultiSelectList onChange={() => alert('onchange happened')} >
```
Comment (emilyrohrbough) [inline packages/terra-list/tests/nightwatch/multi-select-list/OnChangeMultiSelectList.jsx]: Let me know if these updated examples are what you were thinking!

Comment (JakeLaCombe) [inline packages/terra-list/tests/nightwatch/multi-select-list/OnChangeMultiSelectList.jsx]: That'll work

**Comment by mmalaker** [comment]
Functional verification completed. All list examples and test pages display and function correctly.
